Tools/SetupUnityPackage/UnityUtils/FileHelper.cs

```diff
@@ -33,16 +33,22 @@ public static string GetInternalStorage()
 
             // On Android persistentDataPath returns the external folder, where the app may not have permissions to write.
             // we use reflection to call File.getAbsolutePath(currentActivity.getFilesDir())
-            using (var unityPlayer = new AndroidJavaClass("com.unity3d.player.UnityPlayer"))
-            using (var currentActivity = unityPlayer.GetStatic<AndroidJavaObject>("currentActivity"))
-            {
-                var getFilesDir = AndroidJNIHelper.GetMethodID(AndroidJNI.FindClass("android/content/ContextWrapper"), "getFilesDir", "()Ljava/io/File;");
-                var internalFilesDir = AndroidJNI.CallObjectMethod(currentActivity.GetRawObject(), getFilesDir, Array.Empty<jvalue>());
-                var getAbsolutePath = AndroidJNIHelper.GetMethodID(AndroidJNI.FindClass("java/io/File"), "getAbsolutePath", "()Ljava/lang/String;");
+            var path = string.Empty;
 
-                var path = AndroidJNI.CallStringMethod(internalFilesDir, getAbsolutePath, Array.Empty<jvalue>());
-                return path ?? $"/data/data/{Application.identifier}/files";
+            try
+            {
+                using (var unityPlayer = new AndroidJavaClass("com.unity3d.player.UnityPlayer"))
+                using (var currentActivity = unityPlayer.GetStatic<AndroidJavaObject>("currentActivity"))
+                {
+                    var filesDir = currentActivity.Call<AndroidJavaObject>("getFilesDir");
+                    path = filesDir.Call<string>("getAbsolutePath");
+                }
             }
+            catch
+            {
+            }
```

Comment:
This try/catch may not be that useful but
- `getAbsoluteFile` could throw `SecurityException`, which I guess it's pretty rare)
- `GetStatic` may return null if it can't find what's asked, which I can't think of an easy example for currentActivity out of the top of my head

In summary, I think that for precaution is good to be there.